File: Lesson 4 Counting Elements/Missing Integer.py

def solution(A):
    m = len(A)
    a = [0] *m

    for i in A:
        if 0 < i < m+1:
            a[i-1] =1
    try:
        return a.index(0)+1
    except :
        return len(A)+1

File: Lesson 4 Counting Elements/Missing Integer.java
import java.util.*;


class Solution {
    public int solution(int[] A) {
        // write your code in Java SE 11
        boolean flag = false;
        Arrays.sort(A);
        int i = 0;
        for (int s = 1; s<A.length+2 ;s++)
        {
            
            for(;i<A.length;i++)
            {
                if (s == A[i])
                {
                    flag = true;
                    break;
                }
            }
            if (flag ==false)
            {
                return s;
            }
            flag = false;
        }
        //if not found
        return -1;
    }
}